src/ser/mod.rs

```diff
@@ -214,29 +241,26 @@ impl Default for PrettyConfig {
 ///
 /// You can just use `to_string` for deserializing a value.
 /// If you want it pretty-printed, take a look at the `pretty` module.
-pub struct Serializer {
-    output: String,
+pub struct Serializer<W: io::Write> {
```

Comment:
this is great! I was doing the same in [Naga](https://github.com/gfx-rs/naga/blob/1a1f6a47b06c89abb1dff70326c076f1088964a3/src/back/msl.rs#L308) and thinking that it would be cool to get back to Ron and fix it :)